File: src/game/Bot.java
package game;

import action.Action;
import ai.*;
import verification.Verification;

public abstract class Bot extends Player {
    protected MinMax minMax;

    public Bot() {
        super();
        setName("Bot1");
        this.minMax = new MinMax(this, new Evaluation1());
    }
    public Bot (String name, Verification holeVerify, int [][] board) {
        super(name, holeVerify, board);
        this.minMax = new MinMax(this, new Evaluation1());
    }

    public void setMinMax(MinMax minMax) { this.minMax = minMax; }

}

File: projectC/awale/src/game/bot.c
#include "../../headers/game/bot.h"

Bot createBot(char *name)
{
    Bot bot;
    Evaluation evaluation;
    evaluation.evaluate = evaluate_evaluation1;
    bot.player = createPlayer(name);
    bot.minMax = createMinMax(&(bot.player), &(evaluation));

    return bot;
}

void setMinMax(Bot *bot, MinMax minMax)
{
    bot->minMax = minMax;
}

void playBot(Bot *bot, int board[16][3])
{
    Action action = bot->chooseAction(bot);
    printf("pat ici");
    // int nbSeed = action.execute(board);

    // if (nbSeed > 0)
    // {
    //     printf("%s took %d seeds", bot->player.name, nbSeed);
    //     bot->player.nbSeed += nbSeed;
    // }
}
